### PR TITLE
add speed blog on simple steps to more readable/shareable code

### PR DESCRIPTION
**Pull request by davidwyld**
Very quick (30 min) blog which I've been meaning to write on simple steps to writing better code, inspired by some projects I've worked on lately.

### CODE REVIEW

File: _posts/2022-10-20-better-code.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+When you write a function, give it a sensible name, meaning that when reading your code it's easier to tell what that function is doing, just from its name, without having to go and read its source code.
+
+Sometimes a language or community (or even within a project) may have a convention on function names. For example, if the function acquires the value of a variable, its name may begin with `get_`. Look up the conventions or style guide for the language you're using. Following standard practice also makes your code easier to follow.
+
+Function names may also be a good indicator of whether you need to split your function down into smaller logical chunks. If you end up giving a function a very long name e.g. `add_numbers_and_export_them_to_spreadsheet`, then these steps may not logically go together, so should live in different functions.
+
+## 4. Comments?
+Writing comments inline with code is something that most people writing code know about. But is it always the answer?
+Yes, comments can help to explain what's going on in your code and all code should have some.
+But what might be better is to write your code in a way that inherently makes it easier to understand.
```
Comment (ns-rse) [inline _posts/2022-10-20-better-code.md]:
```suggestion
But if you find you are writing large amounts of comments in your code what might be better is to write your code in a way that inherently makes it easier to understand.
```

File: _posts/2022-10-20-better-code.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+If you follow the steps discussed so far, you may find that your code explains itself and comments are only needed in a few places.
+
+## 5. Consider dependencies
+If your code relies on another library, package or toolbox, that's what we call a dependency. Dependencies are needed to run your code, but aren't actually part of your codebase. In order for somone else to use your code (think about openly sharing your research) then they'll need to have those dependencies as well.
+When writing your code, consider which depdendencies are required and make sure that when your code is shared, you also document which other dependencies are required and how to get hold of them.
+
```
Comment (ns-rse) [inline _posts/2022-10-20-better-code.md]: Although I see you may have this planned for a subsequent post given the items listed under *Next Steps?*.

```suggestion
One useful approach to dealing with dependencies is to turn you code into a package which will typically pull in the required dependencies when it is installed.
```

**Review by ns-rse — APPROVED** [review]
Nice post, concise and you've achieved your aim of keeping it language neutral.

Made a couple of very minor comments but it would be fine without these.

**Comment by davidwyld** [comment]
Thanks for the feedback @ns-rse !

**Pull request merged**